### PR TITLE
Fail gracefully on fetching users from gitlab groups

### PR DESCRIPTION
**Pull request by jboeijenga**
## Hey, I just made a Pull Request!

This pull requests makes it easier to see what is going wrong while fetching users from gitlab groups.
Currently the complete refresh of data fails while only a selected number of groups could give an error.
This MR should handle errors like https://github.com/backstage/backstage/issues/20274 more graceful.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by bakito** [comment]
@jboeijenga It would be great, if you could add group information in the 'GraphQL errors' log messages.
This would help finding erroneous groups

File: plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts [hunk 1]
```diff
@@ -245,9 +245,12 @@ export class GitlabOrgDiscoveryEntityProvider implements EntityProvider {
...
       groupRes.matches.push(group);
 
-      const groupUsers = await client.getGroupMembers(group.full_path, [
-        'DIRECT',
-      ]);
+      let groupUsers: PagedResponse<GitLabUser> = { items: [] };
+      try {
+        groupUsers = await client.getGroupMembers(group.full_path, ['DIRECT']);
+      } catch {
+        logger.error(`Failed fetching users for group: ${group.full_path}`);
```
Comment (freben) [inline plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts:252]:
```suggestion
      } catch (e) {
        logger.error(`Failed fetching users for group '${group.full_path}': ${e}`);
```

right?

File: plugins/catalog-backend-module-gitlab/src/lib/client.ts [hunk 2]
```diff
@@ -240,7 +236,11 @@ export class GitLabClient {
         },
       ).then(r => r.json());
       if (response.errors) {
-        throw new Error(`GraphQL errors: ${JSON.stringify(response.errors)}`);
+        throw new Error(
+          `GraphQL errors: ${JSON.stringify(
+            response.errors,
+          )}, for group: ${groupPath}`,
```
Comment (freben) [inline plugins/catalog-backend-module-gitlab/src/lib/client.ts:242]: the caller already adds the group info, no need to add it here too

**Review by alde — APPROVED** [review]

File: plugins/catalog-backend-module-gitlab/src/lib/types.ts [hunk 3]
```diff
@@ -13,9 +13,13 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
```
Comment (freben) [inline plugins/catalog-backend-module-gitlab/src/lib/types.ts:16]: D:

i'm the type of person who keeps re-ADDING these newlines :) Not fond of things being squashed together with the license info

**Review by freben — APPROVED** [review]
OK, but the drawback of this is that it can introduce flakiness for _transient_ problems. For example if you're in a scenario where you get spurious rate limit errors, you'll get a seemingly perfectly valid ingestion happening but where some groups suddenly have no members for some time.

For non-spurious problems though, this makes sense.

**Pull request merged**